src/lib_ccaligner/grammar_tools.cpp

```diff
@@ -200,6 +200,19 @@ std::string getFileData(std::string _fileName)           //returns whole read fi
     std::ifstream infile(_fileName);
     std::string allData = "";
     std::string line;
+
+	char a, b, c;
+	a = infile.get();
+	b = infile.get();
+	c = infile.get();
+	if (a != (char)0xEF || b != (char)0xBB || c != (char)0xBF) {
```

Comment:
Try to avoid C-style cast. Also, maybe you want to use `unsigned char` instead of `char`? since the standard does not specify if `char` is signed or unsigned.